api/models/assetrelationship.go

```diff
@@ -0,0 +1,39 @@
+// Copyright © 2023 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package models
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+func (a *AssetRelationship) ToAsset() (*Asset, error) {
+	if a == nil {
+		return nil, fmt.Errorf("asset relationship is nil")
+	}
+
+	B, err := json.Marshal(a)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal: %v", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("failed to marshal: %w", err)
```

Reply:
I wonder if there is a linter which can catch these things, we should be wrapping everywhere now except in specific instances

Reply:
Looks like its this one:
https://golangci-lint.run/usage/linters/#errorlint

The check is disabled by default but we can enable it using the "errorf" config option.

Reply:
We definitely should in order to make sure we do wrap errors properly.